run_lint.sh

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
+
+START=f142077094608ceb37f7605d2c4f53243be4fb22
+END=HEAD
+
+DIFF=`git diff --name-only $START $END`
+COMMITS=`git log --format=%H $START..$END`
+
+GREP_FILTER="^("
+for d in $COMMITS; do
+  GREP_FILTER+=$d
+  GREP_FILTER+="|"
+done
+
+GREP_FILTER=${GREP_FILTER::-1}
+GREP_FILTER+=")"
```

Comment:
From the `git blame` help:

"       When you are not interested in changes older than version v2.6.18, or changes older than 3 weeks, you can
       use revision range specifiers similar to git rev-list:

           git blame v2.6.18.. -- foo
           git blame --since=3.weeks -- foo

       When revision range specifiers are used to limit the annotation, lines that have not changed since the range
       boundary (either the commit v2.6.18 or the most recent commit that is more than 3 weeks old in the above
       example) are blamed for that range boundary commit.
"

I believe you could save yourself the trouble of doing a big grep, and just filter out all those commits that have `$START` as commit revision.